Marlin/lcdprint_u8g.cpp

```diff
@@ -0,0 +1,98 @@
+/**
+ * @file    lcdprint_u8g.c
+ * @brief   LCD print api for u8glib
+ * @author  Yunhui Fu (yhfudev@gmail.com)
+ * @version 1.0
+ * @date    2016-08-19
+ * @copyright GPL/BSD
+ */
+
+#include "fontutils.h"
+#include "u8g_fontutf8.h"
+
+#include "lcdprint.h"
+
+////////////////////////////////////////////////////////////
+#if ENABLED(DOGLCD)
+
+#if defined(ARDUINO)
+#include <U8glib.h>
+extern U8GLIB *pu8g;
+#define _lcd_write(a) pu8g->print(a)
+#define _lcd_setcursor(col, row) pu8g->setPrintPos((col), (row));
+#else
+#define _lcd_write(a) TRACE ("Write LCD: %c (%d)", (a), (int)(a));
+#define _lcd_setcursor(col, row) TRACE ("Set cursor LCD: (%d,%d)", (col), (row));
+#endif
+
+int
+lcd_glyph_height(void)
+{
+    return u8g_GetFontBBXHeight(pu8g->getU8g());
+    //return u8g_GetFontBBXOffY(pu8g->getU8g());
+}
+
+void
+lcd_moveto (int col, int row)
+{
+    TRACE ("Move to: (%d,%d)", col, row);
+    _lcd_setcursor(col, row);
+}
+
+int
+lcd_print_uchar (wchar_t c, pixel_len_t max_length)
+{
+    if (c < 128) {
+        TRACE ("draw char: regular %d", (int)c);
+        _lcd_write  ((char)c);
+        return u8g_GetFontBBXWidth(pu8g->getU8g());
+    }
+    unsigned int ret;
+    unsigned int x;
+    unsigned int y;
+
+    x = pu8g->getPrintCol();
+    y = pu8g->getPrintRow();
+    TRACE ("uxg_DrawWchar(x=%d,y=%d,maxlen=%d", x, y, max_length);
+    ret = uxg_DrawWchar (pu8g->getU8g(), x, y, c, max_length);
+    TRACE ("u8g->setPrintPos(x=%d + ret=%d,y=%d", x, ret, y);
+    pu8g->setPrintPos (x + ret, y);
+
+    return ret;
+}
+
+int
+lcd_printstr (const char * utf8_str, pixel_len_t max_length)
+{
+    unsigned int ret;
+    unsigned int x;
+    unsigned int y;
```

Comment:
When I post a line note it pertains the line it is posted on. So here I'm referring to the declarations of `ret`, `x`, and `y`.